packages/compiler/src/server/tspconfig/completion.ts

```diff
@@ -80,6 +146,89 @@ export async function provideTspconfigCompletionItems(
         itemsFromEmitter.push(...more);
       }
       return [...itemsFromBuiltIn, ...itemsFromEmitter];
+    } else if (nodePath.length > CONFIG_PATH_LENGTH_FOR_LINTER_LIST && nodePath[0] === "linter") {
+      const extendKeyWord = nodePath[CONFIG_PATH_LENGTH_FOR_LINTER_LIST - 1];
+      const items: CompletionItem[] = [];
+      const linters = await linterProvider.listLibraries(tspConfigFile);
+
+      if (extendKeyWord === "extends") {
+        for (const [name, pkg] of Object.entries(linters)) {
+          // If a ruleSet exists for the linter, add it to the end of the library name.
+          const exports = await pkg.getModuleExports();
+          if (exports?.$linter?.ruleSets !== undefined) {
+            // Below ruleSets are objects rather than arrays
+            for (const [ruleSet] of Object.entries(exports?.$linter?.ruleSets)) {
+              const labelName = `${name}/${ruleSet}`;
+              if (siblings.includes(labelName)) {
+                continue;
+              }
+
+              items.push(
+                getCommonCompetionItem(
+                  labelName,
+                  sourceType,
+                  source,
+                  (await pkg.getPackageJsonData())?.description ?? `Linters from ${labelName}`,
+                  tspConfigPosition,
+                ),
+              );
+            }
+            continue;
+          }
+
+          // If there is no corresponding ruleSet in the library, add the library name directly.
+          if (siblings.includes(name)) {
+            continue;
+          }
+
+          items.push(
+            getCommonCompetionItem(
+              name,
+              sourceType,
+              source,
+              (await pkg.getPackageJsonData())?.description ?? `Linters from ${name}`,
+              tspConfigPosition,
+            ),
+          );
+        }
+      } else {
+        // enable/disable rules
```

Comment:
please check the string explicitly and handle other invalid caes in case of future change